From 234a5e217438ebfeb17555b5ab1d210e4c9820fe Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Tue, 24 Jan 2012 13:54:57 -0500
Subject: Include the default window icon

Since gnome-icon-theme stopped shipping the window icon.

Patch formatted for git by Colin Walters <walters@verbum.org>

https://bugzilla.gnome.org/show_bug.cgi?id=616246
---
 src/Makefile.am     |   6 ++++--
 src/mini-window.png | Bin 0 -> 345 bytes
 src/ui/ui.c         |  12 ++++--------
 src/window.png      | Bin 0 -> 548 bytes
 4 files changed, 8 insertions(+), 10 deletions(-)
 create mode 100644 src/mini-window.png
 create mode 100644 src/window.png

diff --git a/src/Makefile.am b/src/Makefile.am
index cfbce37..4d405bf 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -180,10 +180,12 @@ xml_DATA     = $(xml_in_files:.xml.in=.xml)
 
 @INTLTOOL_SCHEMAS_RULE@
 
-IMAGES=stock_maximize.png stock_minimize.png stock_delete.png
+IMAGES=stock_maximize.png stock_minimize.png stock_delete.png window.png mini-window.png
 VARIABLES=stock_maximize_data $(srcdir)/stock_maximize.png \
           stock_minimize_data $(srcdir)/stock_minimize.png \
-          stock_delete_data $(srcdir)/stock_delete.png
+          stock_delete_data $(srcdir)/stock_delete.png \
+          mini_window_data $(srcdir)/mini-window.png \
+          window_data $(srcdir)/window.png
 
 BUILT_SOURCES = inlinepixbufs.h
 CLEANFILES = inlinepixbufs.h metacity.desktop metacity-wm.desktop org.gnome.metacity.gschema.xml 50-metacity-launchers.xml 50-metacity-navigation.xml 50-metacity-screenshot.xml 50-metacity-system.xml 50-metacity-windows.xml
diff --git a/src/mini-window.png b/src/mini-window.png
new file mode 100644
index 0000000..ce66be1
Binary files /dev/null and b/src/mini-window.png differ
diff --git a/src/ui/ui.c b/src/ui/ui.c
index 33adb0f..78aa353 100644
--- a/src/ui/ui.c
+++ b/src/ui/ui.c
@@ -745,10 +745,8 @@ meta_ui_get_default_window_icon (MetaUI *ui)
                                                    0,
                                                    NULL);
       else
-          default_icon = gtk_icon_theme_load_icon (theme,
-                                                   "gtk-missing-image",
-                                                   META_ICON_WIDTH,
-                                                   0,
+        default_icon = gdk_pixbuf_new_from_inline (-1, window_data,
+                                                   FALSE,
                                                    NULL);
 
       g_assert (default_icon);
@@ -780,10 +778,8 @@ meta_ui_get_default_mini_icon (MetaUI *ui)
                                                    0,
                                                    NULL);
       else
-          default_icon = gtk_icon_theme_load_icon (theme,
-                                                   "gtk-missing-image",
-                                                   META_MINI_ICON_WIDTH,
-                                                   0,
+        default_icon = gdk_pixbuf_new_from_inline (-1, mini_window_data,
+                                                   FALSE,
                                                    NULL);
 
       g_assert (default_icon);
diff --git a/src/window.png b/src/window.png
new file mode 100644
index 0000000..cfbcf38
Binary files /dev/null and b/src/window.png differ
-- 
cgit v0.11.2

